crmsh/ui_resource.py

```diff
@@ -635,76 +639,110 @@ def _get_trace_rsc(self, rsc_id):
+    def _trace_resource(self, context, rsc_id, rsc, dir):
         """Enable RA tracing for a specified resource."""
         op_nodes = rsc.node.xpath('operations/op')
 
         def trace(name):
             for o in op_nodes:
                 if o.get('name') == name:
                     return
-            if not self._add_trace_op(rsc, name, '0'):
+            if not self._add_trace_op(rsc, name, '0', dir):
                 context.fatal_error("Failed to add trace for %s:%s" % (rsc_id, name))
         trace('start')
         trace('stop')
         if xmlutil.is_ms_or_promotable_clone(rsc.node):
             trace('promote')
             trace('demote')
         for op_node in op_nodes:
-            rsc.set_op_attr(op_node, constants.trace_ra_attr, "1")
+            op_node = rsc.set_op_attr(op_node, constants.trace_ra_attr, "1")
+            rsc.set_op_attr(op_node, constants.trace_dir_attr, dir)
 
-    def _trace_op(self, context, rsc_id, rsc, op):
+    def _trace_op(self, context, rsc_id, rsc, op, dir):
         """Enable RA tracing for a specified operation."""
         op_nodes = rsc.node.xpath('operations/op[@name="%s"]' % (op))
         if not op_nodes:
             if op == 'monitor':
                 context.fatal_error("No monitor operation configured for %s" % (rsc_id))
-            if not self._add_trace_op(rsc, op, '0'):
+            if not self._add_trace_op(rsc, op, '0', dir):
                 context.fatal_error("Failed to add trace for %s:%s" % (rsc_id, op))
         for op_node in op_nodes:
-            rsc.set_op_attr(op_node, constants.trace_ra_attr, "1")
+            op_node = rsc.set_op_attr(op_node, constants.trace_ra_attr, "1")
+            rsc.set_op_attr(op_node, constants.trace_dir_attr, dir)
 
-    def _trace_op_interval(self, context, rsc_id, rsc, op, interval):
+    def _trace_op_interval(self, context, rsc_id, rsc, op, interval, dir):
         """Enable RA tracing for an operation with the exact interval."""
         op_node = xmlutil.find_operation(rsc.node, op, interval)
         if op_node is None and utils.crm_msec(interval) != 0:
             context.fatal_error("Operation %s with interval %s not found in %s" % (op, interval, rsc_id))
         if op_node is None:
-            if not self._add_trace_op(rsc, op, interval):
+            if not self._add_trace_op(rsc, op, interval, dir):
                 context.fatal_error("Failed to add trace for %s:%s" % (rsc_id, op))
         else:
-            rsc.set_op_attr(op_node, constants.trace_ra_attr, "1")
+            op_node = rsc.set_op_attr(op_node, constants.trace_ra_attr, "1")
+            rsc.set_op_attr(op_node, constants.trace_dir_attr, dir)
 
     @command.completers(compl.primitives, _raoperations)
-    def do_trace(self, context, rsc_id, op=None, interval=None):
-        'usage: trace <rsc> [<op>] [<interval>]'
+    def do_trace(self, context, rsc_id, *args):
+        'usage: trace <rsc> [-a=<op>] [-i=<interval>] [-d=<log-dir>]'
```

Comment:
AFAICS, by introducing these short arguments, it'd kind of break the convention of crmsh commands.

I understand it's for the precise parsing of these optional options and avoiding any confusions. But it seems to me it's still possible for the parsing code to intelligently differentiate a `dir` from a `op` and `interval`, since it won't be a known type of `op` or a timing value, and it should start with `/`.